### PR TITLE
Get arch per device and support CUDA 9.2+ 

### PR DESCRIPTION
**Pull request by leofang**
I noticed the max compute capability used in the NVRTC support is slightly outdated, so I updated the values according to https://en.wikipedia.org/wiki/CUDA#GPUs_supported; I wasn't able to find a single table listing all these from CUDA's documentation, so it'd be worth double checking. 

Also, the global cache `_nvrtc_max_compute_capability` is replaced by a per-device cache to deal with heterogeneous environments (see https://github.com/cupy/cupy/issues/3292#issuecomment-616267757).

### CODE REVIEW

**Comment by leofang** [comment]
https://docs.nvidia.com/cuda/archive/8.0/nvrtc/index.html#group__options

https://docs.nvidia.com/cuda/archive/9.0/nvrtc/index.html#group__options
https://docs.nvidia.com/cuda/archive/9.1/nvrtc/index.html#group__options
https://docs.nvidia.com/cuda/archive/9.2/nvrtc/index.html#group__options

https://docs.nvidia.com/cuda/archive/10.0/nvrtc/index.html#group__options
https://docs.nvidia.com/cuda/archive/10.1/nvrtc/index.html#group__options
https://docs.nvidia.com/cuda/nvrtc/index.html#group__options (currently points to 10.2)

**Comment by leofang** [comment]
Ah, I think we could simplify the changes a bit if 8.0 is to be dropped (#3301).

**Comment by kmaehashi** [comment]
Thank you!
You may have already noticed, but `compute_72` is used for Xavier so we want to avoid using '75' as the default. I think it's better to maintain a list of known Tegra CCs (72, 62, 53) and use it if the device returns it.

**Comment by kmaehashi** [comment]
You can check this page for the full list of CCs.
https://developer.nvidia.com/cuda-gpus

**Comment by leofang** [comment]
> I think it's better to maintain a list of known Tegra CCs (72, 62, 53) and use it if the device returns it.

Good point! Will do.

> You can check this page for the full list of CCs.
> https://developer.nvidia.com/cuda-gpus

Oh great! I've seen this long time ago but totally forgot about it...Thanks!

**Comment by leofang** [comment]
@kmaehashi PTAL

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by leofang** [comment]
Not sure the root of the errors...Let me take a look.

**Comment by leofang** [comment]
Looks like the mock device has a bug? `arch = device.Device().compute_capability` returned `'60'` but it is supposed to be mocked to `'62'`.

https://jenkins.preferred.jp/job/chainer/job/cupy_pr/1301/TEST=cupy-py3,label=mn1-p100/testReport/junit/tests.cupy_tests.cuda_tests.test_compiler/TestNvrtcArch/test_get_arch_cuda9/

**Comment by leofang** [comment]
Ah, the return value of `_get_arch` is already cached when we run other tests...Let me fix it.

**Comment by leofang** [comment]
@kmaehashi e5e3360 should work

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by leofang** [comment]
Some network issue?

**Comment by kmaehashi** [comment]
Let me try again...
pfnCI, test this please.

**Comment by kmaehashi** [comment]
LGTM!
(ignoring failing test known as flaky: https://github.com/chainer/chainer/issues/8551)

**Comment by leofang** [comment]
Thanks!
